src/store/actions/api.actions.js

```diff
@@ -57,7 +67,30 @@ export const willClone = data => ({
   payload: data,
 });
 
-export const fetchAPI = (pathname) => dispatch => {
+export const deleteAPI = (apiName, callback) => dispatch => {
+  dispatch(deleteAPIRequest());
+
+  return client.delete(`apis/${apiName}`)
+    .then(response => {
+      dispatch(deleteAPISuccess());
+      dispatch(openResponseModal({
```

Comment:
I believe it's better to handle DELETE_API_SUCCESS action in reducer. So no need to dispatch another action. And all your logic is in reducers